lib/eex/lib/eex/compiler.ex

```diff
@@ -406,11 +423,19 @@ defmodule EEx.Compiler do
     state.engine.handle_body(buffer)
   end
 
-  defp generate_buffer([{:eof, meta}], _buffer, _scope, state) do
+  defp generate_buffer([{:eof, meta}], _buffer, [{_content, content_meta} | _scope], state) do
+    snippet = code_snippet(state.source, {content_meta.line, content_meta.line}, arrow_padding: 1)
+
+    message = """
+    unexpected end of string, expected a closing '<% end %>'.
```

Comment:
I think in this case, we need to change the message slightly:

```suggestion
    expected a closing '<% end %>' for block expression in EEx
```

Because we are showing the snippet that starts, I think the message should focus on the beginning rather than the end of the string :)